service/dap/server_test.go

```diff
@@ -500,9 +1082,14 @@ func runDebugSessionWithBPs(t *testing.T, client *daptest.Client, launchRequest
 
 	// Program automatically continues to breakpoint or completion
 
-	for _, onBP := range onBreakpoints {
+	for _, onBP := range onBPs {
```

Comment:
IMHO this would be clearer if it was more like withTestProcessArgs in proc_test: a single function pointer gets called here and then if it wants to continue it calls client.ContinueRequest/client.ExpectContinueResponse/client.ExpectStoppedEvent (possibly using a helper function).

However this is a personal opinion, so you can ignore this comment.

Reply:
I do see your point. Thank you for the suggestion. I added a TODO and will take this into consideration next time I have to reshuffle the tests.